src/Symfony/Component/Validator/ExecutionContext.php

```diff
@@ -271,7 +271,15 @@ public function getMetadataFactory()
      */
     private function executeConstraintValidators($value, array $constraints)
     {
+        // Filter constraints with the context group.
+        $filteredConstraints = array();
         foreach ($constraints as $constraint) {
+            if (in_array($this->getGroup(), $constraint->groups)) {
```

Comment:
It's necessary because for embedded constraint with multiple validation groups, the validator is execute for each validation groups.

With this example:

```
$entity = new Entity();
$entity->firstName = array('baz' => 2);

$this->metadata->addPropertyConstraint('firstName', new Collection(
    array(
        'fields' => array(
            'baz' => array(
                new Range(array('min' => 3, 'groups' => array('foo', 'bar'))),
            ),
        ),
    )
));

$violations = $this->validate($entity, null, array('foo', 'bar'));
```

We have two violations but we want only one. 

Is there another way to fix that ?